cpp/src/slice2js/Gen.cpp

```diff
@@ -501,72 +457,46 @@ Slice::JsVisitor::writeConstantValue(
 void
 Slice::JsVisitor::writeDocCommentFor(const ContainedPtr& p, bool includeDeprecated)
 {
-    StringList lines;
-
+    assert(!dynamic_pointer_cast<Operation>(p));
+    CommentPtr comment = p->parseComment(jsLinkFormatter);
+    if (!comment)
     {
-        string comment = p->comment();
-        string::size_type pos = 0;
-        string::size_type nextPos;
-        string line;
-        while ((nextPos = comment.find_first_of('\n', pos)) != string::npos)
-        {
-            line = string(comment, pos, nextPos - pos);
-            pos = line.find_first_not_of(" \t");
-            if (pos != string::npos)
-            {
-                line = line.substr(pos);
-            }
-            lines.push_back(line);
-            pos = nextPos + 1;
-        }
-
-        line = string(comment, pos);
-        pos = line.find_first_not_of(" \t");
-        if (pos != string::npos)
-        {
-            line = line.substr(pos);
-        }
-
-        if (line.find_first_not_of(" \t\n\r") != string::npos)
-        {
-            lines.push_back(line);
-        }
+        // There's nothing to write for this doc-comment.
+        return;
```

Comment:
There is a small issue here, definition marked as deprecated with metadata without a doc comment would be not marked as deprecated in JavaScript because of this early return.

Fixing in a follow up PR.